src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,37 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(beast) {
+    if ((!beast.hidden) && (beast.hidden !== undefined)) {
```

Comment:
![image](https://user-images.githubusercontent.com/64437769/148067614-46d0efdd-6587-47ec-ba38-c09884d6a432.png)

and why do you need to check if it is not undefined?

Reply:
I did it to check does "beast" is herbivore. Becouse only herbivores have "hidden" property. But I agree that the "beast instanceof Herbivore" is more correctly and readable solution. Thank you!